pkg/morph/client/multi.go

```diff
@@ -66,14 +72,12 @@ func (x *multiClient) iterateClients(f func(*Client) error) error {
 		}
 
 		x.clientsMtx.Lock()
-
 		c, cached := x.clients[x.endpoints[i]]
+		x.clientsMtx.Unlock()
 		if !cached {
```

Comment:
did not get: we checked if it is cached here but then check once again in the `createForAddress`